delta-app/src/main/java/io/cdap/delta/store/DraftService.java

```diff
@@ -290,12 +290,16 @@ private TableAssessmentResponse assessTable(SourceTable sourceTable, TableRegist
 
     // fetch detail about the table, then filter out columns that will not be read by the source
     TableDetail detail = tableRegistry.describeTable(db, table);
+    List<String> selectedPrimaryKeys = detail.getPrimaryKey().stream()
```

Comment:
This should show up as a feature problem in the assessment. Something like:

'Column 'name' is part of the primary key for table 'db.table', but is not one of the selected fields to be replicated. This can result in different data at the target than at the source.

Reply:
True. Totally agree. we should distinguish between the real root cause. Will do.